networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/rpc/methods/BeaconBlocksByRootMessageHandler.java

```diff
@@ -15,29 +15,46 @@
 
 import org.apache.logging.log4j.Level;
 import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
-import tech.pegasys.artemis.datastructures.networking.libp2p.rpc.BeaconBlocksMessageRequest;
+import tech.pegasys.artemis.datastructures.networking.libp2p.rpc.BeaconBlocksByRootRequestMessage;
 import tech.pegasys.artemis.networking.p2p.jvmlibp2p.Peer;
 import tech.pegasys.artemis.networking.p2p.jvmlibp2p.rpc.LocalMessageHandler;
 import tech.pegasys.artemis.networking.p2p.jvmlibp2p.rpc.ResponseCallback;
+import tech.pegasys.artemis.storage.ChainStorageClient;
 import tech.pegasys.artemis.util.alogger.ALogger;
 
-public class BeaconBlocksMessageHandler
-    implements LocalMessageHandler<BeaconBlocksMessageRequest, BeaconBlock> {
-  private final ALogger LOG = new ALogger(BeaconBlocksMessageHandler.class.getName());
+public class BeaconBlocksByRootMessageHandler
+    implements LocalMessageHandler<BeaconBlocksByRootRequestMessage, BeaconBlock> {
+  private final ALogger LOG = new ALogger(BeaconBlocksByRootMessageHandler.class.getName());
+
+  private final ChainStorageClient storageClient;
+
+  public BeaconBlocksByRootMessageHandler(final ChainStorageClient storageClient) {
+    this.storageClient = storageClient;
+  }
 
   @Override
   public void onIncomingMessage(
       final Peer peer,
-      final BeaconBlocksMessageRequest message,
+      final BeaconBlocksByRootRequestMessage message,
       final ResponseCallback<BeaconBlock> callback) {
     LOG.log(
         Level.DEBUG,
         "Peer "
             + peer.getPeerId()
-            + " requested BeaconBlocks starting from "
-            + message.getStartSlot()
-            + ".");
-    // TODO Stub
+            + " requested BeaconBlocks with roots: "
+            + message.getBlockRoots());
+    if (storageClient.getStore() != null) {
+      message
+          .getBlockRoots()
+          .forEach(
+              blockRoot -> {
+                final BeaconBlock block = storageClient.getStore().getBlock(blockRoot);
+                if (block != null) {
+                  callback.respond(block);
```

Comment:
Not sure how to interpret this, but [the spec](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/networking/p2p-interface.md#beaconblocksbyroot) makes is sound like we're required to send at least one chunk back.  Not sure what that would be if we have nothing though:

> The response MUST consist of at least one response_chunk and MAY consist of many. Each successful response_chunk MUST contain a single BeaconBlock payload.

Reply:
I tried to test this against Lighthouse but they haven't finished updating their req/resp to v0.9 and with current master we don't seem to detect the end of their responses - dug into that a fair bit and we just don't seem to get any events up from Netty and as far as I can tell Netty is still expecting to get more responses.  When connecting to Artemis we get a very clear close signal (and it's not our code doing the closing, something down in libp2p is).

So I'm inclined to go with the current no-response approach despite the spec until we can get talking to another client and see what they do.